### PR TITLE
Doctors Appointment

### PR DESCRIPTION
**Pull request by brahimdidi**
**Doctor's Appointment** is a real-world practical app that allows a user to log into the site and view the available doctors in order to make an appointment. The user can also view the details of the doctor and make an appointment by providing the specific date and time as well as the description/nature of the appointment. Moreover, the user can view the list of appointments that he/she has made. Functionality is also provided for the user to delete/cancel an appointment.

## Live Demo Link 
[Live Demo Link](https://doc-appointments.netlify.app/)

## Backend App Link
[Backend - API](https://github.com/TracK92/doctor-appointment-api)

## Backend Pull Request
[Backend - Pull Request](https://github.com/TracK92/doctor-appointment-api/pull/27)

## CONTRIBUTORS

👤 **Tracey Kadenyi**

- GitHub: [@TracK92](https://github.com/TracK92)
- LinkedIn: [@Tracey Kadenyi](https://www.linkedin.com/in/Tracey-Kadenyi/)

👤 **Aleksandra Ujvari**

- GitHub: [@digia3d](https://github.com/digia3d)
- LinkedIn: [Aleksandra (Digital Marketing A3D) Ujvari](https://www.linkedin.com/in/aleksandra-ujvari-85235a210/) 
- Twitter: [@ujvari65](https://twitter.com/ujvari65)

👤 **Abby Nyakara**

- GitHub: [@AbbyNyakara](https://github.com/AbbyNyakara)
- Twitter: [@AbigaelNyakara](https://twitter.com/AbbyNyakara)
- LinkedIn: [Abigael Nyakara](https://linkedin.com/in/AbbyNyakara)

👤 **Brahim didi**

- GitHub: [@brahimdidi](https://github.com/brahimdidi)
- Twitter: [@brahimdidi](https://twitter.com/twitterhandle)
- LinkedIn: [@brahimdidi](https://linkedin.com/in/brahimdidi)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,201 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## Preview
+- COMING SOON
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- COMING SOON
```
Comment (Meltrust) [inline README.md:84]: - [ ] You have a deploment that works.  Please, include it here.

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import DrDetail from './components/DoctorDetail/DrDetail';
+import 'react-toastify/dist/ReactToastify.css';
+import ProtectRoutes from './components/ProtectRoutes';
+import Appointments from './components/BookAppointments/Appointments';
+import Display from './components/DisplayAppointments/Display';
+import NewDoctor from './components/Doctors/NewDoctor';
+import Navbar from './components/Navbar/Navbar';
+import About from './components/About';
+
+function App() {
```
Comment (Meltrust) [inline src/App.js:18]: - [ ]  Please, kindly refactor this functional component  to ES6 arrow function, so your code complies with the latest standards.

The function keyword has been soft deprecated all over the software industry.  This means, almost nobody uses it anymore, so much that it's considered non best practice.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry.

File: src/components/ProtectRoutes.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { Navigate, Outlet } from 'react-router-dom';
+
+function ProtectRoutes() {
```
Comment (Meltrust) [inline src/components/ProtectRoutes.js:4]: - [ ]  Please, kindly refactor this functional component  to ES6 arrow function, so your code complies with the latest standards.

The function keyword has been soft deprecated all over the software industry.  This means, almost nobody uses it anymore, so much that it's considered non best practice.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry.

File: src/components/Authentication/User.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import { Navigate } from 'react-router-dom';
+import ClipLoader from 'react-spinners/ClipLoader';
+import { toast } from 'react-toastify';
+import {
+  fetchCreateUser, fetchUsers, selectAll, selectUsers, setCurrentUser, selectCurrentUser,
+  setLoginUser,
+} from '../../Redux/UserReducer';
+import './user.css';
+
+function User() {
```
Comment (Meltrust) [inline src/components/Authentication/User.js:13]:
```suggestion
function User() {- [ ]  Please, kindly refactor this functional component  to ES6 arrow function, so your code complies with the latest standards.

The function keyword has been soft deprecated all over the software industry.  This means, almost nobody uses it anymore, so much that it's considered non best practice.

Refactoring is easy:

```javascript
const App = () => {
  // ...
}
```

## Explanation

The `function` keyword has been soft deprecated in almost all the software industry in favor of ES6 arrow functions.

You can do a search in Vscode for the "function" word, and correct as needed.  Shouldn't take you more than 5-10 min so don't worry.
```

File: README.md [hunk 5]
```diff
@@ -1 +1,201 @@
...
+- [👥 Authors](#authors)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [❓ FAQ](#faq)
+- [📝 License](#license)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 [Doctor's Appointment] <a name="about-project"></a>
```
Comment (Meltrust) [inline README.md:36]: - [ ] I couldn't find an open PR in the backend project to review.  For your next review request, kindly open a PR with all the changes to review.   It should point from dev to main.

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @brahimdidi

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- App is working well✔️
- Nice design ✔️  

You can do it

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js (hunk 2 above)
Comment (AbbyNyakara) [reply src/App.js:18]: We have refactored the code

File: src/components/ProtectRoutes.js (hunk 3 above)
Comment (AbbyNyakara) [reply src/components/ProtectRoutes.js:4]: We have corrected the links

File: README.md (hunk 5 above)
Comment (AbbyNyakara) [reply README.md:36]: Done

**Review by mcrd25 — APPROVED** [review]
Hi  Team, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**